File: tests/indexing/parsing/lang/fixtures/php/Conditionals.php
<?php

function describe(int $score): string
{
    $tier = "";
    if ($score >= 90) {
        $tier = "gold";
        $score = $score - 90;
    } elseif ($score >= 50) {
        $tier = "silver";
    } else {
        $tier = "bronze";
        $score = 0;
    }
    return $tier . ":" . $score;
}

function settle(int $value, int $min, int $max): int
{
    $adjusted = $value;
    if ($value < $min) {
        $deficit = $min - $value;
        $adjusted = $min + intdiv($deficit, 2);
    }
    if ($value > $max) {
        $adjusted = $max;
        return $adjusted - 1;
    }
    return $adjusted;
}

File: tests/indexing/parsing/lang/fixtures/ruby/Conditionals.rb
class Grader
  def tier(score, streak)
    bonus = streak * 2
    if score + bonus >= 90
      grade = "gold"
      reward = score + bonus - 90
      [grade, reward]
    elsif score >= 50
      grade = "silver"
      [grade, streak]
    else
      grade = "bronze"
      penalty = 50 - score
      [grade, penalty]
    end
  end

  def clamp(value, floor, ceil)
    value = floor if value < floor
    value = ceil unless value <= ceil
    span = ceil - floor
    [value, span]
  end

  def describe(flag, count)
    if flag then
      label = "on"
      "#{label}:#{count}"
    else
      "off:#{count * 0}"
    end
  end
end
